client/components/PetAdoptionForm.js

```diff
@@ -0,0 +1,128 @@
+import React, {useState} from "react"
+
+import ErrorList from "./ErrorsList"
+
+const PetAdoptionForm = (props) => {
+  const [errors, setErrors] = useState({})
+  const [adoption, setAdoption] = useState({
+    name: "",
+    phoneNumber: "",
+    email: "",
+    homeStatus: "",
+    adoptablePetId: props.petId,
+  })
+
+  const handleInputChange = (event) => {
+    setAdoption({
+      ...adoption,
+      [event.currentTarget.id]: event.currentTarget.value
+    })
+  }
+
+  const submitHandler = (event) => {
+    event.preventDefault()
+    if (isValid()) {
+      addPetForAdoption()
+    } 
+  }
+
+  const addPetForAdoption = async () => {
+    try {
+      const response = await fetch("/api/v1/adoptions", {
+        method: "POST",
+        headers: new Headers ({
+          "Content-Type": "application/json",
+        }),
+        body: JSON.stringify(adoption)

```

Comment:
basically between 29-30 we could create a new object with the spread operator to avoid mutating a piece of state directly and combine `props.petId` with `adoption`